samples_project/Assets/SampleViewer/Samples/FeatureLayer/FeatureLayer.cs

```diff
@@ -0,0 +1,421 @@
+        var jFeatures = jObject.SelectToken("features").ToArray();
+        var properties = jFeatures[0].SelectToken("properties");
+        //Populate Outfields drop down
+        foreach (var outfield in properties)
+        {
+            if (outfields.Contains("Get All Features"))
+            {
+                var key = outfield.ToString();
+                var props = key.Split(":");
+                outfields.Add(props[0]);
+                var item = Instantiate(OutfieldItem);
+                item.tag = "ToggleItem";
+                ListItems.Add(item.GetComponent<Toggle>());
+                item.GetComponentInChildren<TextMeshProUGUI>().text = props[0];
+                item.transform.SetParent(contentContainer);
+                item.transform.localScale = Vector2.one;
+            }
+            else
+            {
+                outfields.Add("Get All Features");
+                var item = Instantiate(OutfieldItem);
+                item.tag = "ToggleItem";
+                ListItems.Add(item.GetComponent<Toggle>());
+                item.GetComponentInChildren<TextMeshProUGUI>().text = "Get All Features";
+                item.transform.SetParent(contentContainer);
+                item.transform.localScale = Vector2.one;
+            }
+        }
+    }
+
+    private void MoveCamera()
+    {
+        if (FeatureItems.Count != 0)
+        {
+            if (GetAllFeatures)
+            {
+                var cameraLocationComponent = arcGISCamera.gameObject.GetComponent<ArcGISLocationComponent>();
+                var position = new ArcGISPoint(FeatureItems[0].GetComponent<ArcGISLocationComponent>().Position.X,
+                    FeatureItems[0].GetComponent<ArcGISLocationComponent>().Position.Y, 10000, cameraLocationComponent.Position.SpatialReference);
+                cameraLocationComponent.Position = position;
+                cameraLocationComponent.Rotation = new ArcGISRotation(cameraLocationComponent.Rotation.Heading, 0.0,
+                    cameraLocationComponent.Rotation.Roll);
+            }
+            else
+            {
+                var cameraLocationComponent = arcGISCamera.gameObject.GetComponent<ArcGISLocationComponent>();
+                var position = new ArcGISPoint(FeatureItems[StartValue].GetComponent<ArcGISLocationComponent>().Position.X,
+                    FeatureItems[StartValue].GetComponent<ArcGISLocationComponent>().Position.Y, 10000, cameraLocationComponent.Position.SpatialReference);
+                cameraLocationComponent.Position = position;
+                cameraLocationComponent.Rotation = new ArcGISRotation(cameraLocationComponent.Rotation.Heading, 0.0,
+                    cameraLocationComponent.Rotation.Roll);
+            }   
+        }
+    }
+    
+    public void SelectItems()
+    {
+        foreach (var toggle in ListItems)
+        {
+            var item = toggle.GetComponent<ScrollViewItem>();
+            if (GetAllOutfields)
```

Comment:
This can be made shorter.
```suggestion
            if (GetAllOutfields && item.Data.enabled && item.Data.name != "Get All Features")
            {
                    item.Data.enabled = false;
            }
            if (!GetAllOutfields && item.Data.enabled && item.Data.name == "Get All Features")
            {
                    item.Data.enabled = false;
            }
```
Also, I really don't like coupling this logic to hardcoded strings for the label name.

Reply:
This is the only one that has to be hard coded because we are creating an option to "Get all Features". The others are gotten in the query